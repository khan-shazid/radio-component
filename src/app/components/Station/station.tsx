import React, {useState} from 'react';

import styles from './style.module.css';

import RADIO from '../../assets/radio.png';
import { Radio } from '../Radio/radio';
import Header from '../StationHeader/station-header';
import Footer from '../StationFooter/station-footer';
import { StationModel } from '../../models';

interface StationProps {
    stations: StationModel[]
}

const Station = ({ stations }: StationProps): JSX.Element => {
    const [flag, setFlag] = useState<boolean>(false);

    const [selectedStation, setSelectedStation] = useState<StationModel | undefined>(undefined);

    const handleSelection = (station: StationModel) => {
        setSelectedStation(station);
    }

    return (
        <div className={styles.stationContainer}>
            <Header/>
            <ul className={styles.accordionList}>
            {
                stations.map((station) => {
                    return (<Radio item={station} isSelected={selectedStation?.id === station.id} handleSelection={handleSelection} key={station.id}/>)
                })
            }
            </ul>
            <Footer name={selectedStation?.name}/>
        </div>
    );
};

export default Station;

// export default CSSModules(Station, styles, {allowMultiple: true} )
