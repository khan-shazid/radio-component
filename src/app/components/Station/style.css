* {
    box-sizing: border-box;
}
body {
    padding: 0;
    margin: 0;
    border: 0;
    font-family: "Helvetica Neue", sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}



.station-container {
    padding: 0;
    margin: 0;
    border: 0;
    font-family: "Helvetica Neue", sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    max-width: 440px;
    margin: 100px auto;
    box-shadow: 0 34px 20px 0 rgba(0,0,0,.1);
}
header.station-header {
    display: flex;
    width: 100%;
    background: #EDAD61;
    color: #fff;
    font-size: 22px;
    font-weight: 600;
    text-align: center;
    justify-content: space-between;
    align-items: center;
    padding: 30px 20px;
    border-radius: 28px 28px 0 0;
}
.back, .sign-out, .plus, .minus {
    width: 30px;
    height: 30px;
    cursor: pointer;
    background-size: contain !important;
}

.back {
    background: url("../../assets/back-arrow.png") no-repeat center;
}
.sign-out {
    background: url("../../assets/switch.png") no-repeat center;
}
.plus {
    background: url("../../assets/plus.png") no-repeat center;
}
.minus  {
    background: url("../../assets/minus.png") no-repeat center;
}
ul.accordion-list {
    position: relative;
    display: block;
    width: 100%;
    height: auto;
    padding: 20px;
    margin: 0;
    list-style: none;
    background-color: #2A2A35;
}
ul.accordion-list li {
    position: relative;
    display: block;
    width: 100%;
    height: auto;
    padding: 20px 0px;
    margin: 0;
    cursor: pointer;
    color: #A2ABBD;
    border-bottom: 1px solid #464646;
    line-height: 1;
}
ul.accordion-list li:last-child {
    border-bottom: 0;
}
ul.accordion-list li.active h3:after {
    transform: rotate(45deg);
}
ul.accordion-list li h3 {
    font-weight: normal;
    position: relative;
    display: block;
    width: 100%;
    height: auto;
    padding: 0 0 0 0;
    margin: 0;
    font-size: 15px;
    letter-spacing: 0.01em;
    cursor: pointer;
    color: #A2ABBD;
    font-size: 20px;
    display: flex;
    justify-content: space-between;
}
ul.accordion-list li h3 strong {
    font-weight: bold;
    font-size: 24px;
}
ul.accordion-list li div.answer {
    position: relative;
    display: flex;
    width: 100%;
    height: auto;
    margin: 0;
    padding: 0;
    cursor: pointer;
    justify-content: space-between;
    align-items: center;
}
ul.accordion-list li div.answer p {
    position: relative;
    display: block;
    font-weight: 300;
    padding: 10px 0 0 0;
    cursor: pointer;
    line-height: 150%;
    margin: 0 0 15px 0;
    font-size: 14px;
}
.footer {
    background: #22222C;
    text-align: center;
    padding: 36px;
    border-radius: 0 0 28px 28px ;
}
.footer img{
    width: 25%;
}
