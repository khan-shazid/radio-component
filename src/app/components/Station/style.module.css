* {
    box-sizing: border-box;
}
body {
    padding: 0;
    margin: 0;
    border: 0;
    font-family: "Helvetica Neue", sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}



.stationContainer {
    max-width: 440px;
    margin: 100px auto;
    box-shadow: 0 34px 20px 0 rgba(0,0,0,.1);
    border-radius: 0 0 28px 28px ;
}

.stationLogo {
    width: 30px;
    height: 30px;
    cursor: pointer;
    background-size: contain !important;
}

ul.accordionList {
    position: relative;
    display: block;
    width: 100%;
    height: auto;
    padding: 20px;
    margin: 0;
    list-style: none;
    background-color: #2A2A35;
}
