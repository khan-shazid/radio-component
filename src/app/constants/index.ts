export const SAMPLE_DATA = [
    {
      id: 1,
      name: 'Putin FM',
      count: '666',
      imageUrl: ''
    },
    {
      id: 2,
      name: 'Dribble FM',
      count: '1012',
      imageUrl: ''
    },
    {
      id: 3,
      name: 'Doge FM',
      count: '994',
      imageUrl: ''
    },
    {
      id: 4,
      name: 'Ballads FM',
      count: '871',
      imageUrl: ''
    },
    {
      id: 5,
      name: 'Maximum FM',
      count: '1722',
      imageUrl: ''
    }
];
