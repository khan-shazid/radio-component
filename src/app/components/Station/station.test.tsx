import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import { SAMPLE_DATA } from '../../constants';
import Station from './station';

afterEach(cleanup);

const getComponent = () => {
  const baseProps = {
    stations: SAMPLE_DATA,
  };
  return <Station {...baseProps}/>
}

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(getComponent(), div);
});

it('matches with snapshot', () => {
    const tree = renderer.create(getComponent()).toJSON();
    expect(tree).toMatchSnapshot();
});
