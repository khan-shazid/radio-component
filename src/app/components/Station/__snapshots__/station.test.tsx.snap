// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches with snapshot 1`] = `
<div
  className="stationContainer"
>
  <header
    className="stationHeader"
    data-testid="header"
  >
    <div
      className="back"
    />
    <div
      className="title"
    >
      STATIONS
    </div>
    <div
      className="signOut"
    />
  </header>
  <ul
    className="accordionList"
  >
    <li
      className="listItem"
      data-testid="radio"
      onClick={[Function]}
    >
      <h3
        className="listItemTitle"
      >
        <span
          data-testid="radio-title-name"
        >
          Putin FM
        </span>
        <strong
          data-testid="radio-title-count"
        >
          66,6
        </strong>
      </h3>
    </li>
    <li
      className="listItem"
      data-testid="radio"
      onClick={[Function]}
    >
      <h3
        className="listItemTitle"
      >
        <span
          data-testid="radio-title-name"
        >
          Dribble FM
        </span>
        <strong
          data-testid="radio-title-count"
        >
          101,2
        </strong>
      </h3>
    </li>
    <li
      className="listItem"
      data-testid="radio"
      onClick={[Function]}
    >
      <h3
        className="listItemTitle"
      >
        <span
          data-testid="radio-title-name"
        >
          Doge FM
        </span>
        <strong
          data-testid="radio-title-count"
        >
          99,4
        </strong>
      </h3>
    </li>
    <li
      className="listItem"
      data-testid="radio"
      onClick={[Function]}
    >
      <h3
        className="listItemTitle"
      >
        <span
          data-testid="radio-title-name"
        >
          Ballads FM
        </span>
        <strong
          data-testid="radio-title-count"
        >
          87,1
        </strong>
      </h3>
    </li>
    <li
      className="listItem"
      data-testid="radio"
      onClick={[Function]}
    >
      <h3
        className="listItemTitle"
      >
        <span
          data-testid="radio-title-name"
        >
          Maximum FM
        </span>
        <strong
          data-testid="radio-title-count"
        >
          172,2
        </strong>
      </h3>
    </li>
  </ul>
  <div
    className="footer"
  >
    <label
      data-testid="footer-static"
    >
      NOW PLAYING
    </label>
    <h2
      data-testid="footer-props"
    />
  </div>
</div>
`;
